object.c

```diff
@@ -520,7 +520,9 @@ rb_obj_size(VALUE self, VALUE args, VALUE obj)
  *
  *  Yields <i>obj</i> and returns the result.
  *
- *	'my string'.yield_self {|s|s.upcase} #=> "MY STRING"
+ *  require "ostruct"
+ *  user = OpenStruct.new(firstname: "John", lastname: "Doe")
+ *  user.yield_self { |u| "#{u.firstname} #{u.lastname}" } #=> "John Doe" 
```

Comment:
@eregon in my first version I was using example without additional variable, but then decided to split it for shorter strings. Let's try this without this variable, as you mentioned.

@eregon @bogdanvlviv but still even those examples provided by you can be rewritten without usage of `yield_self`. =)